builtin/providers/aws/resource_aws_nat_gateway.go

```diff
@@ -0,0 +1,169 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsNatGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsNatGatewayCreate,
+		Read:   resourceAwsNatGatewayRead,
+		Delete: resourceAwsNatGatewayDelete,
+
+		Schema: map[string]*schema.Schema{
+			"allocation_id": &schema.Schema{
```

Comment:
when looking this over, I was initially confused at the name choice for `allocation_id` instead of something like `elastic_ip_address` or `eip`. However, I see that the AWS Go SDK uses the word `allocation_id` as well. It makes sense to remain consistent with the AWS Go SDK, but do you have any idea why they chose the name `allocation_id`? It is not nearly as intuitive as most of the other names for fields in the aws_resources used throughout Terraform. </2 cents>